cmd/ci-secret-bootstrap/main.go

```diff
@@ -0,0 +1,240 @@
+
+	"github.com/ghodss/yaml"
+	"github.com/sirupsen/logrus"
+
+	coreapi "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	coreclientset "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+
+	"github.com/openshift/ci-tools/pkg/bitwarden"
+	"github.com/openshift/ci-tools/pkg/util"
+)
+
+type options struct {
+	dryRun         bool
+	kubeConfigPath string
+	configPath     string
+	bwUser         string
+	bwPasswordPath string
+
+	bwPassword    string
+	secretClients map[string]secretClient
+	config        config
+}
+
+type secretClient struct {
+	secretsGetter coreclientset.SecretsGetter
+	config        rest.Config
+}
+
+func parseOptions() options {
+	var o options
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	fs.BoolVar(&o.dryRun, "dry-run", true, "Whether to actually create the secrets with oc command")
+	fs.StringVar(&o.kubeConfigPath, "kubeconfig", "", "Path to the kubeconfig file to use for CLI requests.")
+	fs.StringVar(&o.configPath, "config", "", "Path to the config file to use for this tool.")
+	fs.StringVar(&o.bwUser, "bw-user", "", "Username to access BitWarden.")
+	fs.StringVar(&o.bwPasswordPath, "bw-password-path", "", "Path to a password file to access BitWarden.")
+	if err := fs.Parse(os.Args[1:]); err != nil {
+		logrus.WithError(err).Errorf("cannot parse args: '%s'", os.Args[1:])
+	}
+	return o
+}
+
+func (o *options) validateOptions() error {
+	if o.bwUser == "" {
+		return fmt.Errorf("the mandatory arg bw-user is empty")
+	}
+	if o.bwPasswordPath == "" {
+		return fmt.Errorf("the mandatory arg bw-password-path is empty")
+	}
+	if o.configPath == "" {
+		return fmt.Errorf("the mandatory arg config is empty")
+	}
+	return nil
+}
+
+func (o *options) completeOptions() error {
+	bytes, err := ioutil.ReadFile(o.bwPasswordPath)
+	if err != nil {
+		return err
+	}
+	o.bwPassword = strings.TrimSpace(string(bytes))
+
+	kubeConfigs, _, err := util.LoadKubeConfigs(o.kubeConfigPath)
+	if err != nil {
+		return err
+	}
+
+	bytes, err = ioutil.ReadFile(o.configPath)
+	if err != nil {
+		return err
+	}
+	config := config{}
+	err = yaml.Unmarshal(bytes, &config)
+	if err != nil {
+		return err
+	}
+	o.config = config
+
+	return o.validateCompletedOptions(kubeConfigs)
+}
+
+func (o *options) validateCompletedOptions(kubeConfigs map[string]rest.Config) error {
+	o.secretClients = map[string]secretClient{}
+	if o.bwPassword == "" {
+		return fmt.Errorf("found empty file: '%s'", o.bwPasswordPath)
+	}
+	if len(o.config) == 0 {
+		return fmt.Errorf("found empty config")
+	}
+	toMap := map[string]map[string]string{}
+	for i, secretConfig := range o.config {
+		if len(secretConfig.From) == 0 {
+			return fmt.Errorf("found empty 'From'")
+		}
+		if len(secretConfig.To) == 0 {
+			return fmt.Errorf("found empty 'To'")
+		}
+		for key, bwContext := range secretConfig.From {
+			if key == "" {
+				return fmt.Errorf("config[%d].From: empty key is not allowed", i)
```

Comment:
Not `From`, `from` -- the user is giving you a YAML where `from` is the key they incorrectly set